### ISSUE TITLE
Build failure with boost 1.83.0

### ISSUE THREAD

**Issue by loqs** [zcash/zcash#6753]
### Describe the issue
Build failure with boost 1.83.0 which includes https://github.com/boostorg/signals2/commit/7a16fc1405f01e13463b6ce7723f30f40ba41fa4

#### If so, please list the steps to reproduce below:
1.  ./autogen
2.  ./configure --prefix=/usr
3.  make

### Expected behaviour
Build should complete without error.

### Actual behaviour + errors
See below.

### The version of Zcash you were using:
Issue can be reproduced using both 5.6.3 b7af08e90de89bf0d9e7ece54ad391d852eb7f7c and 5.3.0 35186b00928f3ba994f0e66bb234c412cbffc7b6

### Machine specs:
- OS name + version: Arch Linux Rolling release
- CPU:  -
- RAM: -
- Disk size: -
- Disk Type (HD/SDD): -
- Linux kernel version (uname -a): ```Linux arch 6.4.12-1-stable #1 SMP Wed, 23 Aug 2023 18:01:09 +0000 x86_64 GNU/Linux```
- Compiler version (gcc --version):  clang --version
```
clang version 16.0.6
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
- Linker version (ld -v): ```GNU ld (GNU Binutils) 2.41.0```
- Assembler version (as --version):
```
GNU assembler (GNU Binutils) 2.41.0
Copyright (C) 2023 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or later.
This program has absolutely no warranty.
This assembler was configured for a target of `x86_64-pc-linux-gnu'.
```

### Any extra information that might be useful in the debugging process.
```CXXFLAGS='-O2 -Wno-unsafe-buffer-usage -Wno-unused-result'``` is used to suppress other warnings which are converted to errors.  This is not needed with 5.3.0 which also has the issue.

<details>
<summary>Reproduction steps and outputs for 5.6.3:</summary>

```
$ ./autogen.sh
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'build-aux/m4'.
libtoolize: copying file 'build-aux/m4/libtool.m4'
libtoolize: copying file 'build-aux/m4/ltoptions.m4'
libtoolize: copying file 'build-aux/m4/ltsugar.m4'
libtoolize: copying file 'build-aux/m4/ltversion.m4'
libtoolize: copying file 'build-aux/m4/lt~obsolete.m4'
configure.ac:37: installing 'build-aux/ar-lib'
configure.ac:35: installing 'build-aux/compile'
configure.ac:24: installing 'build-aux/config.guess'
configure.ac:24: installing 'build-aux/config.sub'
configure.ac:30: installing 'build-aux/install-sh'
configure.ac:30: installing 'build-aux/missing'
Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'build-aux/m4'.
libtoolize: copying file 'build-aux/m4/libtool.m4'
libtoolize: copying file 'build-aux/m4/ltoptions.m4'
libtoolize: copying file 'build-aux/m4/ltsugar.m4'
libtoolize: copying file 'build-aux/m4/ltversion.m4'
libtoolize: copying file 'build-aux/m4/lt~obsolete.m4'
configure.ac:45: installing 'build-aux/compile'
configure.ac:45: installing 'build-aux/config.guess'
configure.ac:45: installing 'build-aux/config.sub'
configure.ac:28: installing 'build-aux/install-sh'
configure.ac:28: installing 'build-aux/missing'
Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'build-aux/m4'.
libtoolize: copying file 'build-aux/m4/libtool.m4'
libtoolize: copying file 'build-aux/m4/ltoptions.m4'
libtoolize: copying file 'build-aux/m4/ltsugar.m4'
libtoolize: copying file 'build-aux/m4/ltversion.m4'
libtoolize: copying file 'build-aux/m4/lt~obsolete.m4'
configure.ac:672: warning: $as_echo is obsolete; use AS_ECHO(["message"]) instead
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
build-aux/m4/ax_pthread.m4:88: AX_PTHREAD is expanded from...
configure.ac:672: the top level
configure.ac:70: installing 'build-aux/compile'
configure.ac:27: installing 'build-aux/config.guess'
configure.ac:27: installing 'build-aux/config.sub'
configure.ac:37: installing 'build-aux/install-sh'
configure.ac:37: installing 'build-aux/missing'
src/Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
$ ./configure --prefix=/usr
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether /usr/bin/clang++ accepts -g... yes
checking for /usr/bin/clang++ option to enable C++11 features... none needed
checking whether make supports the include directive... yes (GNU style)
checking dependency style of /usr/bin/clang++... gcc3
checking for g++... g++
checking whether the compiler supports GNU Objective C++... no
checking whether g++ accepts -g... no
checking dependency style of g++... gcc3
checking whether /usr/bin/clang++ supports C++17 features with -std=c++17... yes
checking whether std::atomic can be used without link library... yes
checking how to print strings... printf
checking for gcc... /usr/bin/clang
checking whether the compiler supports GNU C... yes
checking whether /usr/bin/clang accepts -g... yes
checking for /usr/bin/clang option to enable C11 features... none needed
checking whether /usr/bin/clang understands -c and -o together... yes
checking dependency style of /usr/bin/clang... gcc3
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by /usr/bin/clang... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from /usr/bin/clang object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /usr/bin/clang supports -fno-rtti -fno-exceptions... yes
checking for /usr/bin/clang option to produce PIC... -fPIC -DPIC
checking if /usr/bin/clang PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/clang static flag -static works... yes
checking if /usr/bin/clang supports -c -o file.o... yes
checking if /usr/bin/clang supports -c -o file.o... (cached) yes
checking whether the /usr/bin/clang linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... /usr/bin/clang++ -std=c++17 -E
checking for ld used by /usr/bin/clang++ -std=c++17... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the /usr/bin/clang++ -std=c++17 linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for /usr/bin/clang++ -std=c++17 option to produce PIC... -fPIC -DPIC
checking if /usr/bin/clang++ -std=c++17 PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/clang++ -std=c++17 static flag -static works... yes
checking if /usr/bin/clang++ -std=c++17 supports -c -o file.o... yes
checking if /usr/bin/clang++ -std=c++17 supports -c -o file.o... (cached) yes
checking whether the /usr/bin/clang++ -std=c++17 linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for ar... /usr/bin/ar
checking for ranlib... /usr/bin/ranlib
checking for strip... /usr/bin/strip
checking for git... /usr/bin/git
checking for rustc... /usr/bin/rustc
checking for cargo... /usr/bin/cargo
checking for cxxbridge... /usr/bin/cxxbridge
checking for ccache... no
checking for xgettext... /usr/bin/xgettext
checking for hexdump... /usr/bin/hexdump
checking for readelf... /usr/bin/readelf
checking for c++filt... /usr/bin/c++filt
checking whether C++ compiler accepts -Werror... yes
checking whether C++ compiler accepts -g... yes
checking whether C++ compiler accepts -fwrapv... yes
checking whether C++ compiler accepts -msse4.2... yes
checking whether C++ compiler accepts -msse4.1... yes
checking whether C++ compiler accepts -mavx -mavx2... yes
checking whether C++ compiler accepts -msse4 -msha... yes
checking for SSE4.2 intrinsics... yes
checking for SSE4.1 intrinsics... yes
checking for AVX2 intrinsics... yes
checking for SHA-NI intrinsics... yes
checking whether C++ compiler accepts -march=armv8-a+crc+crypto... no
checking for ARM CRC32 intrinsics... no
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking whether byte ordering is bigendian... no
checking how to run the C preprocessor... /usr/bin/clang -E
checking whether /usr/bin/clang is Clang... yes
checking whether Clang needs flag to prevent "argument unused" warning when linking with -pthread... no
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for /usr/bin/clang++ -std=c++17 options needed to detect all undeclared functions... none needed
checking whether strerror_r is declared... yes
checking whether strerror_r returns char *... yes
checking for __attribute__((visibility))... yes
checking for __attribute__((dllexport))... no
checking for __attribute__((dllimport))... no
checking for library containing clock_gettime... none required
checking whether C++ compiler accepts -fPIC... yes
checking whether C++ compiler accepts -Wformat... yes
checking whether C++ compiler accepts -Wformat-security... yes
checking whether C++ compiler accepts -Wstack-protector... yes
checking whether C++ compiler accepts -fstack-protector-all... yes
checking whether C++ compiler accepts -Wthread-safety-analysis... yes
checking whether C++ preprocessor accepts -D_FORTIFY_SOURCE=2... yes
checking whether C++ preprocessor accepts -U_FORTIFY_SOURCE... yes
checking whether the linker accepts -Wl,-z,relro... yes
checking whether the linker accepts -Wl,-z,now... yes
checking whether C++ compiler accepts -fPIE... yes
checking whether the linker accepts -pie... yes
checking for endian.h... yes
checking for sys/endian.h... no
checking for byteswap.h... yes
checking for stdio.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for strings.h... (cached) yes
checking for sys/types.h... (cached) yes
checking for sys/stat.h... (cached) yes
checking for sys/select.h... yes
checking for sys/prctl.h... yes
checking for library containing getaddrinfo_a... none required
checking for library containing inet_pton... none required
checking whether strnlen is declared... yes
checking whether le16toh is declared... yes
checking whether le32toh is declared... yes
checking whether le64toh is declared... yes
checking whether htole16 is declared... yes
checking whether htole32 is declared... yes
checking whether htole64 is declared... yes
checking whether be16toh is declared... yes
checking whether be32toh is declared... yes
checking whether be64toh is declared... yes
checking whether htobe16 is declared... yes
checking whether htobe32 is declared... yes
checking whether htobe64 is declared... yes
checking whether bswap_16 is declared... yes
checking whether bswap_32 is declared... yes
checking whether bswap_64 is declared... yes
checking whether __builtin_clz is declared... no
checking whether __builtin_clzl is declared... no
checking whether __builtin_clzll is declared... no
checking for MSG_NOSIGNAL... yes
checking for gmtime_r... yes
checking for visibility attribute... yes
checking for thread_local support... yes
checking for fdatasync... yes
checking for F_FULLFSYNC... no
checking for O_CLOEXEC... yes
checking for __builtin_prefetch... yes
checking for _mm_prefetch... yes
checking for strong getauxval support in the system headers... no
checking for weak getauxval support in the compiler... yes
checking for Berkeley DB C++ headers... default
checking for main in -ldb_cxx-6.2... yes
checking for boostlib >= 1.77 (107700)... yes
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
checking whether the Boost::Filesystem library is available... yes
checking for exit in -lboost_filesystem... yes
checking whether the Boost::Program_Options library is available... yes
checking for exit in -lboost_program_options... yes
checking whether the Boost::Thread library is available... yes
checking for exit in -lboost_thread... yes
checking whether the Boost::Chrono library is available... yes
checking for exit in -lboost_chrono... yes
checking whether the Boost::Unit_Test_Framework library is available... yes
checking for dynamic linked boost test... yes
checking for EVENT... yes
checking for EVENT_PTHREADS... yes
checking for ZMQ... yes
checking whether C++ compiler accepts -fno-strict-aliasing... yes
checking whether C++ compiler accepts -Wno-builtin-declaration-mismatch... no
checking whether to build bitcoind... yes
checking whether to build utils (zcash-cli zcash-tx)... yes
checking whether to build libraries... no
checking if ccache should be used... no
checking if wallet should be enabled... yes
checking if mining should be enabled... yes
checking whether to build test_bitcoin... yes
checking whether to reduce exports... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating doc/man/Makefile
config.status: creating src/test/buildenv.py
config.status: creating qa/pull-tester/tests_config.ini
config.status: creating src/config/bitcoin-config.h
config.status: executing depfiles commands
config.status: executing libtool commands
=== configuring in src/secp256k1 (/build/zcash/src/zcash/src/secp256k1)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr'  'CXX=/usr/bin/clang++' 'CXXFLAGS=-O2 -Wno-unsafe-buffer-usage -Wno-unused-result' 'CC=/usr/bin/clang' '--disable-shared' '--with-pic' '--enable-benchmark=no' '--enable-module-recovery' --cache-file=/dev/null --srcdir=.
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for gcc... /usr/bin/clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether /usr/bin/clang accepts -g... yes
checking for /usr/bin/clang option to enable C11 features... none needed
checking whether /usr/bin/clang understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of /usr/bin/clang... gcc3
checking dependency style of /usr/bin/clang... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by /usr/bin/clang... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from /usr/bin/clang object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /usr/bin/clang supports -fno-rtti -fno-exceptions... yes
checking for /usr/bin/clang option to produce PIC... -fPIC -DPIC
checking if /usr/bin/clang PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/clang static flag -static works... yes
checking if /usr/bin/clang supports -c -o file.o... yes
checking if /usr/bin/clang supports -c -o file.o... (cached) yes
checking whether the /usr/bin/clang linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking if /usr/bin/clang supports -Werror=unknown-warning-option... yes
checking if /usr/bin/clang supports -std=c99 -pedantic -Wno-long-long -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef... yes
checking if /usr/bin/clang supports -Wno-overlength-strings... yes
checking if /usr/bin/clang supports -Wall... yes
checking if /usr/bin/clang supports -Wno-unused-function... yes
checking if /usr/bin/clang supports -Wextra... yes
checking if /usr/bin/clang supports -Wcast-align... yes
checking if /usr/bin/clang supports -Wcast-align=strict... no
checking if /usr/bin/clang supports -Wconditional-uninitialized... yes
checking if /usr/bin/clang supports -fvisibility=hidden... yes
checking for valgrind support... 
checking for x86_64 assembly availability... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libsecp256k1.pc
config.status: creating src/libsecp256k1-config.h
config.status: executing depfiles commands
config.status: executing libtool commands

Build Options:
  with external callbacks = no
  with benchmarks         = no
  with tests              = yes
  with coverage           = no
  with examples           = no
  module ecdh             = yes
  module recovery         = yes
  module extrakeys        = yes
  module schnorrsig       = yes

  asm                     = x86_64
  ecmult window size      = 15
  ecmult gen prec. bits   = 4

  valgrind                = no
  CC                      = /usr/bin/clang
  CPPFLAGS                =  
  SECP_CFLAGS             = -O2  -std=c99 -pedantic -Wno-long-long -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wno-overlength-strings -Wall -Wno-unused-function -Wextra -Wcast-align -Wconditional-uninitialized -fvisibility=hidden 
  CFLAGS                  = -g -O2
  LDFLAGS                 = 
=== configuring in src/univalue (/build/zcash/src/zcash/src/univalue)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr'  'CXX=/usr/bin/clang++' 'CXXFLAGS=-O2 -Wno-unsafe-buffer-usage -Wno-unused-result' 'CC=/usr/bin/clang' '--disable-shared' '--with-pic' '--enable-benchmark=no' '--enable-module-recovery' --cache-file=/dev/null --srcdir=.
checking whether make supports nested variables... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking whether make supports the include directive... yes (GNU style)
checking for gcc... /usr/bin/clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether /usr/bin/clang accepts -g... yes
checking for /usr/bin/clang option to enable C11 features... none needed
checking whether /usr/bin/clang understands -c and -o together... yes
checking dependency style of /usr/bin/clang... gcc3
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by /usr/bin/clang... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from /usr/bin/clang object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /usr/bin/clang supports -fno-rtti -fno-exceptions... yes
checking for /usr/bin/clang option to produce PIC... -fPIC -DPIC
checking if /usr/bin/clang PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/clang static flag -static works... yes
checking if /usr/bin/clang supports -c -o file.o... yes
checking if /usr/bin/clang supports -c -o file.o... (cached) yes
checking whether the /usr/bin/clang linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking whether the compiler supports GNU C++... yes
checking whether /usr/bin/clang++ accepts -g... yes
checking for /usr/bin/clang++ option to enable C++11 features... none needed
checking dependency style of /usr/bin/clang++... gcc3
checking how to run the C++ preprocessor... /usr/bin/clang++ -E
checking for ld used by /usr/bin/clang++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the /usr/bin/clang++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for /usr/bin/clang++ option to produce PIC... -fPIC -DPIC
checking if /usr/bin/clang++ PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/clang++ static flag -static works... yes
checking if /usr/bin/clang++ supports -c -o file.o... yes
checking if /usr/bin/clang++ supports -c -o file.o... (cached) yes
checking whether the /usr/bin/clang++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating pc/libunivalue.pc
config.status: creating pc/libunivalue-uninstalled.pc
config.status: creating univalue-config.h
config.status: executing depfiles commands
config.status: executing libtool commands
Fixing libtool for -rpath problems.

Options used to compile and link:
  with wallet   = yes
  with zmq      = yes
  with test     = yes
  use asm       = yes
  sanitizers    = 
  debug enabled = no
  gprof enabled = no
  werror        = yes

  target os     = linux
  build os      = 

  CC            = /usr/bin/clang
  CFLAGS        = -g -O2
  CPPFLAGS      =   -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS
  CXX           = /usr/bin/clang++ -std=c++17
  CXXFLAGS      =  -g  -Wformat -Wformat-security -Wstack-protector -fstack-protector-all -Wthread-safety-analysis  -Werror  -O2 -Wno-unsafe-buffer-usage -Wno-unused-result -fwrapv -fno-strict-aliasing
  LDFLAGS       = -pthread  -Wl,-z,relro -Wl,-z,now -pie  
  ARFLAGS       = cr

$ make
Making all in src
make[1]: Entering directory '/build/zcash/src/zcash/src'
  GEN      rust/gen/include/rust/blake2b.h
  GEN      rust/gen/include/rust/ed25519.h
  GEN      rust/gen/include/rust/equihash.h
  GEN      rust/gen/include/rust/streams.h
  GEN      rust/gen/include/rust/bridge.h
make  all-recursive
make[2]: Entering directory '/build/zcash/src/zcash/src'
make[3]: Entering directory '/build/zcash/src/zcash/src'
  CXX      zcashd-bitcoind.o
  CXX      libbitcoin_server_a-sendalert.o
  CXX      libbitcoin_server_a-addrman.o
  CXX      libbitcoin_server_a-addrdb.o
  CXX      libbitcoin_server_a-alert.o
  CXX      libbitcoin_server_a-asyncrpcoperation.o
  CXX      libbitcoin_server_a-asyncrpcqueue.o
  CXX      libbitcoin_server_a-bloom.o
  CXX      libbitcoin_server_a-chain.o
  CXX      libbitcoin_server_a-checkpoints.o
  CXX      libbitcoin_server_a-deprecation.o
  CXX      libbitcoin_server_a-experimental_features.o
  CXX      libbitcoin_server_a-httprpc.o
  CXX      libbitcoin_server_a-httpserver.o
  CXX      libbitcoin_server_a-init.o
In file included from init.cpp:28:
In file included from ./main.h:20:
In file included from ./net.h:32:
In file included from /usr/include/boost/signals2/signal.hpp:19:
In file included from /usr/include/boost/function.hpp:30:
In file included from /usr/include/boost/function/detail/prologue.hpp:18:
/usr/include/boost/function/function_base.hpp:651:14: error: static_cast from 'obj_ptr_t' (aka 'void *') to 'void (*)(bool, const CBlockIndex *)' is not allowed
      return static_cast<const Functor*>(type_result.members.obj_ptr);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/boost/function/function_base.hpp:657:40: note: in instantiation of function template specialization 'boost::function_base::target<void (bool, const CBlockIndex *)>' requested here
      if (const F* fp = this->template target<F>())
                                       ^
/usr/include/boost/signals2/detail/signal_template.hpp:530:46: note: in instantiation of function template specialization 'boost::function_base::contains<void (bool, const CBlockIndex *)>' requested here
            if((*it)->slot().slot_function().contains(slot))
                                             ^
/usr/include/boost/signals2/detail/signal_template.hpp:222:11: note: in instantiation of function template specialization 'boost::signals2::detail::signal_impl<void (bool, const CBlockIndex *), boost::signals2::optional_last_value<void>, int, std::less<int>, boost::function<void (bool, const CBlockIndex *)>, boost::function<void (const boost::signals2::connection &, bool, const CBlockIndex *)>, boost::signals2::mutex>::do_disconnect<void (bool, const CBlockIndex *)>' requested here
          do_disconnect(slot, is_group());
          ^
/usr/include/boost/signals2/detail/signal_template.hpp:718:19: note: in instantiation of function template specialization 'boost::signals2::detail::signal_impl<void (bool, const CBlockIndex *), boost::signals2::optional_last_value<void>, int, std::less<int>, boost::function<void (bool, const CBlockIndex *)>, boost::function<void (const boost::signals2::connection &, bool, const CBlockIndex *)>, boost::signals2::mutex>::disconnect<void (bool, const CBlockIndex *)>' requested here
        (*_pimpl).disconnect(slot);
                  ^
init.cpp:2021:36: note: in instantiation of function template specialization 'boost::signals2::signal<void (bool, const CBlockIndex *), boost::signals2::optional_last_value<void>, int, std::less<int>, boost::function<void (bool, const CBlockIndex *)>, boost::function<void (const boost::signals2::connection &, bool, const CBlockIndex *)>>::disconnect<void (bool, const CBlockIndex *)>' requested here
        uiInterface.NotifyBlockTip.disconnect(BlockNotifyGenesisWait);
                                   ^
1 error generated.
make[3]: *** [Makefile:5904: libbitcoin_server_a-init.o] Error 1
make[3]: Leaving directory '/build/zcash/src/zcash/src'
make[2]: *** [Makefile:9163: all-recursive] Error 1
make[2]: Leaving directory '/build/zcash/src/zcash/src'
make[1]: *** [Makefile:2901: all] Error 2
make[1]: Leaving directory '/build/zcash/src/zcash/src'
make: *** [Makefile:647: all-recursive] Error 1
```
</details>

I do not know if this is an issue with zcash or boost.
@antonio-rojas @fmhess

**Comment by str4d** [comment]
Thanks for letting us know! This will become relevant once we migrate to Boost 1.83.0.

So you are aware, the only supported build pathway is `./zcutil/build.sh`, which uses pinned dependencies. We currently pin and require Boost 1.82.0:
https://github.com/zcash/zcash/blob/4c32e3b3e3e0f576cf0c37aa260a5a9d1619ec3b/depends/packages/boost.mk#L1-L2

**Comment by foutrelis** [comment]
There is a relevant issue at https://github.com/boostorg/signals2/issues/68. The signals2 documentation [shows examples](https://www.boost.org/doc/libs/1_83_0/doc/html/signals2/tutorial.html#id-1.3.35.4.5) where function pointers are used. However, function names used to work so it's not clear from where it's best to address this (zcash or boost).

**Comment by str4d** [comment]
Fixed by https://github.com/boostorg/function/commit/7ca2310b15e387258e97e4cd16887f5ee4906b28; I will include that patch when I bump our dependencies.

**Comment by str4d** [comment]
Fixed in #6758.
